samples/matter/common/src/bridge/bridge_manager.cpp

```diff
@@ -165,48 +129,48 @@ CHIP_ERROR BridgeManager::AddDevice(BridgedDevice *device)
 	return CHIP_ERROR_NO_MEMORY;
 }
 
-BridgedDevice *BridgeManager::GetBridgedDevice(uint16_t endpoint, const EmberAfAttributeMetadata *attributeMetadata,
-					       uint8_t *buffer)
+BridgeManager::DevicePtr *
+BridgeManager::GetBridgedDevice(uint16_t index, const EmberAfAttributeMetadata *attributeMetadata, uint8_t *buffer)
 {
-	if (endpoint >= kMaxBridgedDevices) {
+	if (index >= kMaxBridgedDevices) {
 		return nullptr;
 	}
 
-	BridgedDevice *device = sBridgeManager.mBridgedDevices[endpoint];
-	if ((device == nullptr) || !buffer || !attributeMetadata) {
+	DevicePtr *device = const_cast<DevicePtr *>(&(sBridgeManager.mDevicesMap[index]->mDevice));
```

Comment:
```suggestion
	BridgedDevice *device = sBridgeManager.mDevicesMap[index]->mDevice->get();
```
Otherwise, I cannot see how `device` could be null. Also, where is it checked that `mDevicesMap` contains `index`?

Reply:
Yep, originally I wanted to use a reference but then though we needed to handle the case when the provided arguments were invalid (buffer and attributeMetadata), but these are not used currently, and maybe won't be. I guess with `Contains()` taking care of the valid index we can drop the GetBridgedDevice() method and use the map directly to avoid a mess with uniqe pointers.